mirrord/layer/src/file/ops.rs

```diff
@@ -444,6 +450,124 @@ pub(crate) fn xstat(
     Detour::Success(response)
 }
 
+/// Logic for the `libc::statx` function.
+/// See [manual](https://man7.org/linux/man-pages/man2/statx.2.html) for reference.
+///
+/// # Warning
+///
+/// Due to backwards compatibility on the [`mirrord_protocol`] level, we use [`XstatRequest`] to get
+/// the remote file metadata.
+/// Because of this, we're not able to fill all field of the [`statx`] structure. Missing fields
```

Comment:
maybe we should zero the buffer before starting to write to it?

Reply:
Inb4 apps that don't check for that flag and rely on zeroing...

Reply:
Example
> If a filesystem does not support a field or if it has an
       unrepresentable value (for instance, a file with an exotic type),
       then the mask bit corresponding to that field will be cleared in
       stx_mask even if the user asked for it and a dummy value will be
       filled in for compatibility purposes if one is available (e.g., a
       dummy UID and GID may be specified to mount under some
       circumstances).

Reply:
Ok, I'm convinced :+1: